nntrainer/tensor/tensor.cpp

```diff
@@ -249,6 +291,15 @@ bool Tensor::operator==(const Tensor &rhs) const {
     } else if (getDataType() == Tdatatype::QINT8) {
       return *std::dynamic_pointer_cast<CharTensor>(itensor) ==
              *std::dynamic_pointer_cast<CharTensor>(rhs.itensor);
+    } else if (getDataType() == Tdatatype::BCQ) {
+#ifdef ENABLE_BIQGEMM
+      return *std::dynamic_pointer_cast<BCQTensor>(itensor) ==
+             *std::dynamic_pointer_cast<BCQTensor>(rhs.itensor);
+#else
+      throw std::invalid_argument(
+        "Error: enable-biqgemm is not activated. "
+        "Enable only if your system supports BiQGEMM.");
+#endif
```

Comment:
How about `ENABLE_BCQ` ? it is not actually 'BiQGEMM' stuff I guess

Reply:
Thank you for sharing your thoughts! To explain, there are two reasons for this.
1. It is designed BCQTensor can be used only when BiQGEMM is available. If BiQGEMM is not available, a user cannot utilize BCQTensor.
2. In meson.build, `enable-biqgemm` checks for the BiQGEMM library.

```
if get_option('enable-biqgemm')
  # check if BiQGEMM directory exist. otherwise, throw an error
  fs = import('fs')
  if fs.is_dir('../BiQGEMM')
    extra_defines += '-DENABLE_BIQGEMM=1'
    biqgemm_inc = include_directories('../BiQGEMM')
  else
    error ('BiQGEMM cannot be enabled without BiQGEMM library.')
  endif
endif
```

Hope this clarifies!

Reply:
Should it be coupled? Can't we decouple BCQTensor and BiQGEMM?

Reply:
Thanks for the explanation!
You have a point, but just wanted to make sure again that BCQ and BiQGEMM are quite different.

Reply:
> Should it be coupled? Can't we decouple BCQTensor and BiQGEMM?

You're right, it can be decoupled! Regarding this issue, I discussed this with @jijoongmoon the other day and we've decided to have BCQTensor can only be used when BiQGEMM is available.

Reply:
> To me, determining it with the word 'BiQGEMM' sounds quite unclear.

I get the point that BCQ and BiQGEMM are different. As @EunjuYang mentioned, it can be decoupled (BCQTensor can be created regardless of the presence of BiQGEMM). However, @jijoongmoon and I decided to restrict the use of BCQ-type by BiQGEMM availability. We can discuss what would be the best option!
Also, you're right about the use of BiQGEMM is arch-dep & type-dep! Current dot implementation is targeted only for full-precision computation. Half-precision computation support will be introduced in the later PR :)

Reply:
Got it.